service/history/replication/executable_task_tool_box.go

```diff
@@ -60,5 +61,7 @@ type (
 		EventSerializer          serialization.Serializer
 		DLQWriter                DLQWriter
 		HistoryEventsHandler     eventhandler.HistoryEventsHandler
+		WorkflowCache            wcache.Cache
```

Comment:
we need to make sure host level workflow cache is always enabled. 
I will check if we can deprecate shard level cache code path in next release.